src/interfaces/queries/query.ts

```diff
@@ -0,0 +1,5 @@
+import { IQuery } from './query.interface';
+
+export class Query<T> implements IQuery {
+  resultType$f9fbca36!: T;
```

Comment:
Shouldn't it be `resultType: T;` ?  right ?
```suggestion
  resultType: T;
```

Reply:
No, this field is made unique intentionally so no one should be surprised while creating a query like 
```
class GetApiResponse extends Query<Response> {
  resultType: string;
  }
  
  const response = queryBus.exec(new GetApiResponse({resultType: 'json'}))
  ```
  in such case, the type of the response variable would be string instead of `Response`.
  That's why @Dyostiq added a unique suffix to that necessary field.

Reply:
Thanks for the reply 😉👍